### ISSUE TITLE
chaperone-procedure doesn't preserve struct predicates

### ISSUE THREAD

**Issue by samth** [samth/pycket#193]
Here's a small program:

```racket
#lang racket/kernel

(#%require racket/private/define
           racket/private/small-scheme
           racket/private/more-scheme)

(define-values (struct:keyword-procedure mk-kw-proc keyword-procedure?
                                         keyword-procedure-ref keyword-procedure-set!)
  (make-struct-type 'keyword-procedure #f 4 0 #f
                    (list (cons prop:checked-procedure #t))
                    (current-inspector)
                    #f
                    '(0 1 2 3)))

(define-values (struct:okp make-optional-keyword-procedure okp? okp-ref okp-set!)
  (make-struct-type 'procedure
                    struct:keyword-procedure
                    1 0 #f
                    (list)
                    (current-inspector) 0))


(define v5 (make-optional-keyword-procedure #f #f null '(#:x) #f))

(define cv2 (chaperone-procedure v5 void))

(display (keyword-procedure? cv2))
```

This prints #t in Racket, #f in Pycket.

**Comment by samth** [comment]
@rjnw This is the (or at least a) mbta bug.

**Comment by sabauma** [comment]
Fixed by 24c571d29cb0e19565f678710fc990d3b72c3f1a
